integrations/repo_test.go

```diff
@@ -74,3 +75,26 @@ func TestViewRepo1CloneLinkAuthorized(t *testing.T) {
 	sshURL := fmt.Sprintf("%s@%s:user2/repo1.git", setting.RunUser, setting.SSH.Domain)
 	assert.Equal(t, sshURL, link)
 }
+
+func testRepoLinksNoLogin(repoPath string, t *testing.T) {
```

Comment:
It would be nice to combine these tests with the existing tests above, but perhaps that can be a separate PR.